internal/telemetry/client.go

```diff
@@ -340,35 +350,58 @@ func (c *client) Count(namespace Namespace, name string, value float64, tags []s
 // sent to the backend. Requests are sent in the background. Must be called
 // with c.mu locked
 func (c *client) flush() {
-	submissions := make([]*Request, 0, len(c.requests)+1)
+	submissions := make([]*Request, 0, len(c.requests)+2)
+
+	// copy over requests so we can do the actual submission without holding
+	// the lock. Zero out the old stuff so we don't leak references
+	for i, r := range c.requests {
```

Comment:
We have to populate the submissions slice before adding metrics events so that the sequence id's are in the right order.  Sequence id represents the order in which telemetry events happened. `newRequest` increments the client's sequence id, and `newRequest` for metrics is only called inside the flush function. 

system tests were failing because the submissions array had out-of-order sequence id's, for example:
`[6, 1, 2, 3, 4, 5]`. 
Telemetry system tests does not expect the events to arrive in a strict order, but it will assert that out-of-order events arrive within a certain timeframe of each other (0.3 seconds), and in this example event 6 was arriving way before event 5.